lib/marked.js

```diff
@@ -555,7 +553,7 @@ inline.pedantic = merge({}, inline.normal, {
 
 inline.gfm = merge({}, inline.normal, {
   escape: replace(inline.escape)('])', '~|])')(),
-  url: XRegExp.cache('^((?:[A-Za-z][A-Za-z\\d-.+]*:(?:\\/{1,3}|[\\pL\\d%])|www\\d{0,3}[.]|[\\pL\\d.\\-]+[.]\\pL{2,4}\\/)(?:\\[[\\da-f:]+\\]|[^\\s`!()\\[\\]{;:\'",<>?«»“”‘’]|[`!\\[\\]{;:\'",<>?«»“”‘’](?=[^\\s()<>])|\\((?:[^\\s()<>]|(?:\\([^\\s()<>]+\\)))*\\))+)', ''),
+  url: /^((?:[A-Za-z][A-Za-z\d.+-]*:(?:\/{1,3}|[\p{L}\d%])|www\d{0,3}[.]|[\p{L}\d.-]+[.]\p{L}{2,4}\/)(?:\[[\da-f:]+\]|[^\s`!()\[\]\{;:'",<>?«»“”‘’]|[`!\[\]\{;:'",<>?«»“”‘’](?=[^\s()<>])|\((?:[^\s()<>]|(?:\([^\s()<>]+\)))*\))+)/u,
```

Comment:
I don't really like that regex:
1. It does not match subdomains without protocol (`community.mattermost.com`)
2. It auto-matches anything when adding a random protocol (e.g. `foo:community.mattermost.com`)
3. it auto-matches anything when starting the string with `www.`
4. it allows for unsupported characters in `URI`s (or `IRI`s)

I would vote to replace that with a RFC compliant regex

Reply:
The first 3 of those are intentional. The 4th isn't intentional, but it works well enough.

1. We don't want to match absolutely everything that looks like a URL because that'll end up autolinking a lot of things that weren't intended to be links.
2. This is also intended, but there's more code down the line that checks the protocols against a list of allowed protocols which contains both standard ones (http, mailto, ftp) and an option for admins to allow other custom ones.
3. This is intended since we wanted to support people just typing www links.
4. That's not intended, but I based this on the regex that GitHub used for their autolinking, so I trust it even if it's incorrect.

Regardless though, we shouldn't change that now. These regexes have caused a lot of problems in the past, and it's much safer to leave them as-is even if the validation isn't 100% correct.

Also, other regexes we may come across, particularly in RFCs, are likely built with the assumption that an entire string either is or isn't a URL. This one will be different and potentially more complicated because we're trying to identify URLs within other text

Reply:
In that case it might be ok .. I will investigate a bit more, but to not block this we can go on with it